### PR TITLE
JsonSerializer: Added option to provide a custom serialization 'replacer' 

### PR DESCRIPTION
**Pull request by sailingKieler**
During my work on the generation tracing stuff, I temporarily used such a hook to modify the AST serialization.

I would love to have it in the code base, as it gives flexibility for use case specific serializations without the need to subclass the JsonSerializer. What do you think?

### CODE REVIEW

**Comment by spoenemann** [comment]
Isn't it sufficient to override the `replacer` method of JsonSerializer?

**Comment by sailingKieler** [comment]
> Isn't it sufficient to override the replacer method of JsonSerializer?

You would have to create an additional class und register that.
With this hook one can just call the default serializer like
```
const json = jsonSerializer.serialize(document.parseResult.value, {
    replacer: (key, value: unknown, defaultReplacer) => {
        if (key === 'foo') {
            return value.toString();
        } else {
            return defaultReplacer(key, value);
        }
    }
);
```
That was super helpful for me.

File: packages/langium/src/serializer/json-serializer.ts [hunk 1]
```diff
@@ -53,7 +54,10 @@ export class DefaultJsonSerializer implements JsonSerializer {
     }
 
     serialize(node: AstNode, options?: JsonSerializeOptions): string {
-        return JSON.stringify(node, (key, value) => this.replacer(key, value, options?.refText), options?.space);
+        const defaultReplacer = (key: string, value: unknown) => this.replacer(key, value, options?.refText);
```
Comment (spoenemann) [inline packages/langium/src/serializer/json-serializer.ts:57]: While we're at it, could you change the third parameter of the `replacer` method to the JsonSerializeOptions object so we can extend it more easily in the future?

**Review by spoenemann — COMMENTED** [review]
Ok the change is not obtrusive, 👍 for shipping it

File: packages/langium/src/serializer/json-serializer.ts (hunk 1 above)
Comment (sailingKieler) [reply packages/langium/src/serializer/json-serializer.ts:57]: I had that in mind, too, need to sync with #886 (generator) as there will be a conflict.

**Review by msujew — APPROVED** [review]
Looks good to me 👍

**Pull request merged**